### DISCUSSION TITLE
Cannot read RTMP handshake response

### DISCUSSION THREAD

**Discussion by elcapitan26** [bluenviron/mediamtx#312] [Questions]
## Which version are you using?

docker aler9/rtsp-simple-server:latest

## Question

First off, this project is superb and I want to thank you. I am successfully using the software with RTSP and now trying out the RTMP features. I can confirm that 1935 is open on the container and rtmpEnabled: yes is configured. When I attempt to push a stream using FFmpeg:

ffmpeg -re -stream_loop -1 -i ../in/test.mp4 -c copy -f flv rtmp://IPADDRESS:1935/mystream1

I get the following:

Cannot read RTMP handshake response
rtmp://IPADDRESS:1935/mystream1: End of file

Any guidance would be appreciated and thanks again.

**Comment by aler9** [comment]
hello, i wasn't able to replicate the issue - i started the server in this way

```
docker run --rm -it --network=host -e RTSP_RTMPENABLE=true aler9/rtsp-simple-server
```

and published with

```
ffmpeg -re -i sample_4_3.mp4 -c copy -f flv rtmp://localhost:1935/mystream1
```

The issue may depend on the file, on the ffmpeg version, on the rtsp-simple-server version (make sure that the local image is updated), on the ports (maybe something else is using 1935). You have to try to make it work and then identify the exact element that is causing the problem.

**Comment by elcapitan26** [comment]
Alessandro, I appreciate. I now have the service running. In case anyone else runs into a problem I had to run the container locally using:

docker run --rm -it -e RTSP_RTMPENABLE=yes -e RTSP_PROTOCOLS=tcp -p 8554:8554 -p 1935:1935 aler9/rtsp-simple-server

[a comment by an automated account is not shown]
